server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileTree.java

```diff
@@ -12,16 +12,20 @@
 import org.apache.lucene.search.Collector;
 import org.opensearch.search.profile.ContextualProfileBreakdown;
 import org.opensearch.search.profile.ProfileResult;
+import org.opensearch.search.profile.Timer;
 
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * This class returns a list of {@link ProfileResult} that can be serialized back to the client in the concurrent execution.
  *
  * @opensearch.internal
  */
 public class ConcurrentQueryProfileTree extends AbstractQueryProfileTree {
+    protected Map<Long, LinkedList<Timer>> threadToRewriteTimers = new ConcurrentHashMap<>();
```

Comment:
It is used only in `ConcurrentQueryProfiler` and should belong there, also note that this structure is *not thread safe*, the value (which is linked list) must be safe under concurrent modifications (which is not the case right now).

Reply:
Hi @reta, thank you for the review. I've moved the map to `ConcurrentQueryProfiler.java`. I understand **LinkedList** is not thread safe. But correct me if I'm wrong, this map structure has thread ID as key, so there is no concurrent access for each LinkedList here. So I don't think we need a thread safe data structure for the `Timer` LinkedList.

Reply:
> So I don't think we need a thread safe data structure for the `Timer` LinkedList.

Yes, I agree with you, the thread ID as a key should eliminate this concern, thank you